app/models/report.rb

```diff
@@ -4,24 +4,44 @@ class Report
   attr_reader :exercise_logs, :pain_logs, :pt_sessions, :pains, :exercises, :body_parts
 
   def initialize(args)
-    @filter_params = args
-    @exercise_logs ||= query_logs('exercise_logs')
-    @pain_logs ||= query_logs('pain_logs')
-    @pt_sessions ||= query_logs('pt_sessions')
-    @pains ||= @filter_params[:user].pains
-    @exercises ||= @filter_params[:user].exercises
-    @body_parts ||= @filter_params[:user].body_parts
+    @exercise_logs = args[:exercise_logs]
+    @pain_logs = args[:pain_logs]
+    @pt_sessions = args[:pt_sessions]
+    @pains = args[:pains]
+    @exercises = args[:exercises]
+    @body_parts = args[:body_parts]
   end
 
-  private
+  class << self
+    def build_report(filter_params) # rubocop:disable Metrics/MethodLength
+      @filter_params = filter_params
```

Comment:
having this instance var here feels wrong to me, but I don't want to have to pass it as an arg 3 times.